### DISCUSSION TITLE
chaining the time resolution

### DISCUSSION THREAD

**Discussion by akariwrynn** [pybamm-team/PyBaMM#2343] [General]
Hi,
I would like to measure the voltage and the current at very low time domains for impedance spec. However, after some time voltage seems to be constant for small time changes like:
V(t=0.00097889)
Out[33]: array(4.18440764)

V(t=0.000980453)
Out[34]: array(4.18440764)

V(t=0.000982017)
Out[35]: array(4.18440764)

V(t=0.000983581)
Out[36]: array(4.18440764)
V(t=0.000984363)
Out[39]: array(4.18440787)
V(t=0.000985927)
Out[37]: array(4.18522475)

V(t=0.000985145)
Out[38]: array(4.18522475)

How can I increase the time resolution or fix this problem?

Thanks.

**Comment by YannickNoelStephanKuhn** [comment]
Could you please post the way you invoke the solver? The simplest way to increase the time resolution depends on that.

**Reply by akariwrynn** [reply]
Sure,
i=0
while i< len(Array): # Simulation
    
   params['Current function [A]'] = deneme(amplitude,freq[i])
   sim=pybamm.Simulation(model,parameter_values=params)
   sim.solve(Array[i]) #solving the simulation for the time domain that we specified
   solution=sim.solution
   V=solution["Terminal voltage [V]"]
   C=solution["Current [A]"]
   for e in Array[i]:
       Voltage_gen.append(V(t=e))
       Current_gen.append(C(t=e))
   i+=1
freq is the frequencies that I want to look and deneme is the function that turned to function to sin function and finally array is the time required for each frequency.
Thanks.

**Comment by YannickNoelStephanKuhn** [comment]
This should work:
```
dt_max_in_seconds = np.min(np.abs(Array[1:] - Array[:-1]))
sim.solve(Array[i], dt_max=dt_max_in_seconds * model.timescale_eval)
```

**Reply by akariwrynn** [reply]
Hi again,
It says 'DFN' object has no attribute 'timescale_eval' so, what should I do?

Sorry for bothering too much

**Reply by Tyssel** [reply]
You may try to replace  `model.timescale_eval` by
```python
pybamm.ParameterValues().evaluate(model.timescale)
```

**Comment by YannickNoelStephanKuhn** [comment]
Seems like the documentation is outdated then: https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/solvers/casadi_solver.py#L47
Seems like one of the developers needs to help here.

**Reply by akariwrynn** [reply]
So we wait then

**Reply by rtimms** [reply]
`dt_max` is already in seconds, so assuming `Array` is in seconds you can just do 

```python3
dt_max = np.min(np.abs(Array[1:] - Array[:-1]))
sim.solve(Array[i], dt_max=dt_max)
```

The default is `dt_max = 0.01* model.timescale_eval` and `model.timescale_eval` only exists once the simulation has been set up.

I can see why this is confusing though. We should improve the docs.
